src/main/java/com/epam/ta/reportportal/core/events/handler/item/TestItemAutoAnalysisRunner.java

```diff
@@ -0,0 +1,81 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.ta.reportportal.core.events.handler.item;
+
+import com.epam.ta.reportportal.core.analyzer.auto.AnalyzerService;
+import com.epam.ta.reportportal.core.analyzer.auto.LogIndexer;
+import com.epam.ta.reportportal.core.analyzer.auto.impl.AnalyzerUtils;
+import com.epam.ta.reportportal.core.events.activity.item.TestItemFinishedEvent;
+import com.epam.ta.reportportal.core.events.handler.ConfigurableEventHandler;
+import com.epam.ta.reportportal.core.launch.GetLaunchHandler;
+import com.epam.ta.reportportal.entity.ItemAttribute;
+import com.epam.ta.reportportal.entity.item.TestItem;
+import com.epam.ta.reportportal.entity.launch.Launch;
+import com.epam.ta.reportportal.ws.model.project.AnalyzerConfig;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+
+/**
+ * Run auto analyzer for finished test item with immediateAA attribute.
+ *
+ * @author <a href="mailto:andrei_piankouski@epam.com">Andrei Piankouski</a>
+ */
+@Component
+public class TestItemAutoAnalysisRunner implements
+    ConfigurableEventHandler<TestItemFinishedEvent, Map<String, String>> {
+
+  protected static final String IMMEDIATE_AUTO_ANALYSIS = "immediateAA";
+
+  private final AnalyzerService analyzerService;
+
+  private final LogIndexer logIndexer;
+
+  private final GetLaunchHandler getLaunchHandler;
+
+  public TestItemAutoAnalysisRunner(AnalyzerService analyzerService, LogIndexer logIndexer,
+      GetLaunchHandler getLaunchHandler) {
+    this.analyzerService = analyzerService;
+    this.logIndexer = logIndexer;
+    this.getLaunchHandler = getLaunchHandler;
+  }
+
+  @Override
+  public void handle(TestItemFinishedEvent testItemFinishedEvent,
+      Map<String, String> projectConfig) {
+    if (analyzerService.hasAnalyzers() && isContainAttribute(testItemFinishedEvent.getTestItem())) {
+      final AnalyzerConfig analyzerConfig = AnalyzerUtils.getAnalyzerConfig(projectConfig);
+      TestItem testItem = testItemFinishedEvent.getTestItem();
+      logIndex(testItem, testItemFinishedEvent.getProjectId(), analyzerConfig);
+      Launch launch = getLaunchHandler.get(testItem.getLaunchId());
+      analyzerService.runAnalyzers(launch, List.of(testItem.getItemId()), analyzerConfig);
+      logIndex(testItem, testItemFinishedEvent.getProjectId(), analyzerConfig);
+    }
+  }
+
+  private void logIndex(TestItem testItem, Long projectId, AnalyzerConfig config) {
+    logIndexer.indexItemsLogs(projectId, testItem.getLaunchId(), List.of(testItem.getItemId()),
+        config);
+  }
+
+  private boolean isContainAttribute(TestItem testItem) {
+    Optional<ItemAttribute> first = testItem.getAttributes().stream()
+        .filter(it -> IMMEDIATE_AUTO_ANALYSIS.equals(it.getKey())).findFirst();
+    return first.isPresent() && Boolean.parseBoolean(first.get().getValue());
+  }
```

Comment:
You can rewrite it to use anyMatch like this:
```
    return testItem.getAttributes().stream()
        .anyMatch(it -> IMMEDIATE_AUTO_ANALYSIS.equals(it.getKey()) && Boolean.parseBoolean(it.getValue()));
```

Also the method name should be changed to something like containsImmediateAutoAnalysisAttribute

Reply:
Fixed